ostd/src/task/task/mod.rs

```diff
@@ -195,8 +201,7 @@ impl Task {
         // `current_task()` still holds a strong reference, so nothing is destroyed at this point,
         // neither is the kernel stack.
         drop(self);
-
-        schedule();
+        scheduler::dequeue();
```

Comment:
The name dequeue is inaccurate. Internally, putting a task to sleep also dequeues the task from  its runqueue. So I think it should be called `scheduler::exit_current`.